### PR TITLE
HDDS-8322. [Snapshot] Created various config for Snapdiff and moved few configs from OzoneConfigKeys to OMConfigKeys

### PR DESCRIPTION
**Pull request by hemantk-12**
## What changes were proposed in this pull request?
This change is to create various SnapDiff related config as mentioned in the Jira: [HDDS-8322](https://issues.apache.org/jira/browse/HDDS-8322).

Also moved [these config keys](https://github.com/apache/ozone/blob/2a826133d681e3ed8a789c77b6c6f8c622e4c743/hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/OzoneConfigKeys.java#L594-L617) from [OzoneConfigKeys](https://github.com/apache/ozone/blob/2a826133d681e3ed8a789c77b6c6f8c622e4c743/hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/OzoneConfigKeys.java#L38) to [OmConfigKeys](https://github.com/apache/ozone/blob/184bfc21189583f857280933270f134dcac68879/hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/OMConfigKeys.java#L30) because there are related to OM not Ozone as whole system.

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-8322

## How was this patch tested?
Existing unit tests.

### CODE REVIEW

File: hadoop-hdds/rocksdb-checkpoint-differ/pom.xml [hunk 1]
```diff
@@ -68,11 +68,21 @@ https://maven.apache.org/xsd/maven-4.0.0.xsd">
       <groupId>org.apache.ozone</groupId>
       <artifactId>hdds-test-utils</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.apache.ozone</groupId>
+      <artifactId>ozone-common</artifactId>
+      <version>${ozone.version}</version>
```
Comment (adoroszlai) [inline hadoop-hdds/rocksdb-checkpoint-differ/pom.xml:74]: Please don't add dependency in `hadoop-hdds` for anything in `hadoop-ozone`.  Move code, if necessary.

Comment (hemantk-12) [reply hadoop-hdds/rocksdb-checkpoint-differ/pom.xml]: Removed `ozone-common` dependency and moved configs back to `OzoneConfigKeys` for now.

File: hadoop-hdds/common/src/main/resources/ozone-default.xml [hunk 2]
```diff
@@ -3655,13 +3655,88 @@
...
+      Default wait time returned to client to wait before retrying snap diff request.
+      Uses millisecond by default when no time unit is specified.
+    </description>
+  </property>
+
+  <property>
+    <name>ozone.om.snapshot.diff.max.jobs.purge.per.task</name>
+    <value>1000</value>
+    <tag>OZONE, OM</tag>
+    <description>
```
Comment (prashantpogde) [inline hadoop-hdds/common/src/main/resources/ozone-default.xml:3708]: Isn't it too high a default value ?

Comment (hemantk-12) [reply hadoop-hdds/common/src/main/resources/ozone-default.xml:3735]: Reduced it to 100.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshot.java [hunk 3]
```diff
@@ -161,9 +160,13 @@ private void init() throws Exception {
...
     conf.set(OMConfigKeys.OZONE_DEFAULT_BUCKET_LAYOUT,
         bucketLayout.name());
-    conf.setBoolean(OzoneConfigKeys.OZONE_OM_SNAPSHOT_FORCE_FULL_DIFF,
+    conf.setBoolean(OMConfigKeys.OZONE_OM_SNAPSHOT_FORCE_FULL_DIFF,
         forceFullSnapshotDiff);
     conf.setEnum(HDDS_DB_PROFILE, DBProfile.TEST);
+    conf.setTimeDuration(
+        OMConfigKeys.OZONE_OM_SNAPSHOT_DIFF_JOB_DEFAULT_WAIT_TIME,
+        TimeUnit.SECONDS.toMillis(1),
+        TimeUnit.MILLISECONDS);
```
Comment (smengcl) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshot.java:169]: Could this 1s diff job wait time in the test be a source of flakiness?

**Review by smengcl — APPROVED** [review]
Thanks @hemantk-12 for doing the clean up. overall lgtm

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshot.java (hunk 3 above)
Comment (hemantk-12) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshot.java:169]: No, it will just take more time.

Comment (smengcl) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshot.java:169]: got it.

**Pull request merged**